src/index.ts

```diff
@@ -34,21 +37,63 @@ export class ChainOps {
     }
   }
 
-  async getGasPrice(blockNumber?: number) {
+  async getGasPrice (blockNumber?: number) {
     return oracle.getGasPrice(this.getEndpoint('ORACLE_URL'), blockNumber)
   }
 
-  getEndpoint(endpointName: string): string {
-    //@ts-ignore
+  getEndpoint (endpointName: string): string {
+    // @ts-ignore
     if (!this.config[endpointName] || this.config[endpointName].length === 0) {
       throw new Error(endpointName + ' endpoint not defined')
     }
 
-    //@ts-ignore
+    // @ts-ignore
     return this.config[endpointName]
   }
 
-  async subscribe(subConfig: any) {
+  async getOptimisticBalance (wallet: EthAddress, tokenContract: EthAddress) {
+    if (!wallet) {
+      throw new Error('wallet is required')
+    }
+    if (!tokenContract) {
+      throw new Error('tokenContract is required')
+    }
+    const creds = await this.getCreds()
+
+    return watcher.getOptimisticBalance(
+      this.getEndpoint('SUBSCRIPTIONS_ENDPOINT'),
+      creds,
+      wallet,
+      tokenContract
+    )
+  }
+
+  async logOptimisticPending (executionId: string, tokenContract: EthAddress, senderAddress: EthAddress, tokenAmount: string) {
```

Comment:
❔ Would this benefit from a docstring?  It's not clear to me what purpose this function serves (or even if it's meant to be part of the public interface).